Dockerfile

```diff
@@ -1,20 +1,14 @@
 # Copyright 2021 Synology Inc.
 
 ############## Build stage ##############
-FROM golang:1.13.6-alpine as builder
+FROM golang:1.19-alpine as builder
 LABEL stage=synobuilder
 
 RUN apk add --no-cache alpine-sdk
 WORKDIR /go/src/synok8scsiplugin
-COPY go.mod .
-RUN go mod download
-
-COPY Makefile .
-
+COPY . .
```

Comment:
This is a bad idea. You should separate COPY layer for Dockerfile best practices.

Reply:
This is only the build image which is dropped afterwards, so it died not matter.
The final image only copies the built binary from this image.

Reply:
I added a .dockerignore to skip unwanted files from the docker build

Reply:
Build cache will be used with go packages, so I beleive this should be as is

Reply:
Still, does not impact the final image, as the cached layers are in the builder image.
Anyway, I changed it

Reply:
Btw. You are using alpine:latest as source image for the final image.
You wont be able to reuse cached layers anyway, as latest is most of the time a new image. 
Also not Dockerfile best practices 🤨